crates/task-impls/src/quorum_proposal_recv/mod.rs

```diff
@@ -125,21 +131,51 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions>
         event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
         event_receiver: Receiver<Arc<HotShotEvent<TYPES>>>,
     ) {
-        if let HotShotEvent::QuorumProposalRecv(proposal, sender) = event.as_ref() {
-            match handle_quorum_proposal_recv(
-                proposal,
-                sender,
-                &event_sender,
-                &event_receiver,
-                self,
-            )
-            .await
-            {
-                Ok(()) => {
-                    self.cancel_tasks(proposal.data.view_number()).await;
+        match event.as_ref() {
+            HotShotEvent::QuorumProposalRecv(proposal, sender) => {
+                if self.consensus.read().await.cur_view() > proposal.data.view_number()
+                    || self.cur_view > proposal.data.view_number()
```

Comment:
I think we can but this is effectively the same since view change happens exactly once for each view we we enter and always emits one view change event.  So by handling that event we save one read lock aquire